core/wiki/macros/export.tid

```diff
@@ -1,33 +1,46 @@
 title: $:/core/macros/export
 tags: $:/tags/Macro
 
-\define exportButtonFilename(baseFilename)
-$baseFilename$$(extension)$
-\end
-
-\define exportButton(exportFilter:"[!is[system]sort[title]]",lingoBase,baseFilename:"tiddlers")
 \whitespace trim
-<$vars hint={{{ [<__lingoBase__>addsuffix[Hint]get[text]] }}} caption={{{  [<__lingoBase__>addsuffix[Caption]get[text]] }}}>
-<span class="tc-popup-keep"><$button popup=<<qualify "$:/state/popup/export">> tooltip=<<hint>> aria-label=<<caption>> class=<<tv-config-toolbar-class>> selectedClass="tc-selected">
-<$list filter="[<tv-config-toolbar-icons>match[yes]]">
-{{$:/core/images/export-button}}
-</$list>
-<$list filter="[<tv-config-toolbar-text>match[yes]]">
-<span class="tc-btn-text"><$text text=<<caption>>/></span>
-</$list>
-</$button></span></$vars><$reveal state=<<qualify "$:/state/popup/export">> type="popup" position="below" animate="yes">
-<div class="tc-drop-down">
-<$set name="count" value={{{ [subfilter<__exportFilter__>count[]] }}}>
-<$list filter="[all[shadows+tiddlers]tag[$:/tags/Exporter]]">
-<$list filter="[<currentTiddler>has[condition]subfilter{!!condition}limit[1]] ~[<currentTiddler>!has[condition]then[true]]" variable="ignore">
-<$button class="tc-btn-invisible">
-<$action-sendmessage $message="tm-download-file" $param=<<currentTiddler>> exportFilter=<<__exportFilter__>> filename={{{ [<__baseFilename__>addsuffix{!!extension}] }}}/>
-<$action-deletetiddler $tiddler=<<qualify "$:/state/popup/export">>/>
-<$transclude field="description"/>
-</$button>
-</$list>
-</$list>
-</$set>
-</div>
+
+<!-- Not used, only there for backwards compatibility -->
+\function exportButtonFilename(baseFilename) [<baseFilename>] [<extension>] +[join[]]
+
+\procedure exportButton(exportFilter:"[!is[system]sort[title]]",lingoBase,baseFilename:"tiddlers")
+<$let hint={{{ [<lingoBase>addsuffix[Hint]get[text]] }}} caption={{{  [<lingoBase>addsuffix[Caption]get[text]] }}}>
+	<span class="tc-popup-keep">
+		<$button popup=<<qualify "$:/state/popup/export">>
+			tooltip=<<hint>>
+			aria-label=<<caption>>
+			class=<<tv-config-toolbar-class>>
+			selectedClass="tc-selected"
+		>
+			<% if [<tv-config-toolbar-icons>match[yes]] %>
+				{{$:/core/images/export-button}}
+			<% endif %>
+			<% if [<tv-config-toolbar-text>match[yes]] %>
+				<span class="tc-btn-text"><$text text=<<caption>>/></span>
+			<% endif %>
+		</$button>
+	</span>
+</$let>
+<$reveal state=<<qualify "$:/state/popup/export">> type="popup" position="below" animate="yes">
+	<div class="tc-drop-down">
+		<$set name="count" value={{{ [subfilter<exportFilter>count[]] }}}>
+			<$list filter="[all[shadows+tiddlers]tag[$:/tags/Exporter]]">
+				<% if [<currentTiddler>has[condition]subfilter{!!condition}limit[1]] ~[<currentTiddler>!has[condition]then[true]] %>
```

Comment:
In this line the first part Never run! as the `[<currentTiddler>has[condition]subfilter{!!condition}limit[1]]` always return nothing! This issue is also in the original code!

I tested on tiddlywiki.com

1. `[<currentTiddler>has[condition]subfilter{!!condition}limit[1]]`  return nothing
2. `[subfilter{!!condition}limit[1]]` return the first output

Reply:
If you suspect an issue here, we will need to make some tests first. The condition-field is a possibility to enable / disable export buttons in the dropdown. If the tiddler has no condition-field, the button should be visible.

Reply:
@kookma .. I did some tests and the condition works as expected. It was introduced with commit: https://github.com/TiddlyWiki/TiddlyWiki5/commit/1d7091e637462dcda8c28d660ca42c073f647043 to remove the `.tid` exporter if more than 1 tiddler is listed. 

As soon as the `condition` subfilter returns a value, the element will be shown. See testcase: 

```
title: test-filter
condition: [{!!x}match[test]]
description: test condition-field
extension: .json
x: testXXX
```

![image](https://github.com/user-attachments/assets/2fe5d2e7-ec7b-4ddc-9b3a-1c21c7633a9a)

Reply:
Thank you for clarification.